.github/workflows/check-build-publish.yaml

```diff
@@ -0,0 +1,106 @@
+name: Check, build and publish
+
+on:
+  push:
+    branches:
+      - main
+
+concurrency: 
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  checks:
+    name: Test and lint
+    runs-on: ubuntu-22.04
+    permissions:
+      checks: write
+    steps:
+      - name: Install build dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y libssl-dev pkg-config protobuf-compiler libprotobuf-dev
+
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup rust toolchain
+        uses: actions-rs/toolchain@v1
+        with: # seems to ignore rust-toolchain but it should be supported
+          toolchain: "1.76.0"
+          components: rustfmt, clippy
+
+      - name: Check code formatting
+        uses: actions-rs/cargo@v1
+        with:
+          command: fmt
+          args: --check
+      
+      - name: Clippy linting
+        uses: giraffate/clippy-action@v1
```

Comment:
alternatively, we could add `reviewdog`, though either way i'd add `clippy-flags -- -Dwarnings`

Reply:
I don't have strong feelings on this. The action-rs workflows are maintainer-less at the moment but they can  surface issues into github's action UI  and even go so far as create issues in your repository. This would be esspecially helpful for CVEs. The action-rs clippy integration causes issues when running on a fork but this action behaves as expected and appears to be better maintained.

Does running cargo clippy from the cli has the same effect?

Reply:
I'd have to see the output logs to be sure, to be honest.

Reply:
Modified the clippy command to effectively run: `--all-features --all-targets - -D warnings`